### ISSUE TITLE
Cannot access roboRIO with Domain Name

### ISSUE THREAD

**Issue by tom131313** [Mechanical-Advantage/AdvantageScope#21]
Using advantagescope-win-x64-v2.0.0-beta-1.exe

I understand now that until 2023 I cannot use Connect to Robot because that is only for the new NT4 and Live Monitoring so if I don't touch that function, I am able to download files from the roboRIO using the static IP address 10.42.37.2. If I touch that function, it hangs in Searching and I cannot access the roboRIO even using the static IP address.

I wanted to access the roboRIO download files using http://roborio-4237-frc.local but that does not work (it works for all other programs such as browser and PuTTY). Error says to check connections.

I tried to diagnose the problem with Network Monitor but I couldn't see what is the trouble (I'm not real good at reading the network traffic).

**Comment by jwbonner** [comment]
I'd just like to confirm what you're seeing with the static IP. Is this accurate?

With IP set to "10.42.37.2":
- Open AdvantageScope > "Download Logs..." > Works correctly
- Open AdvantageScope > "Connect to Robot" > "Download Logs..." > Doesn't work?

For the domain name, you probably need to use "roborio-4237-frc.local" instead of with "http://". HTTP is specifically for web connections, which is not useful here.

**Comment by tom131313** [comment]
Right. That static IP works for download. I tried domain name with and without the http://, with and without .local, and with and without a / at the end. I couldn't find in a packet what name was attempting to be resolved so I tried all of my best guesses.

Apparently the connect to robot shouldn't work yet as I do not have the 2023 WPILib installed (waiting 6 more weeks).

The only thing I can think of is the program times out and retries before the DNS resolves the name - it does take a long time to do so (several seconds and variable). Something appears to be in a 30 second loop but I can't tell what. 30 seconds probably is a good timeout period for DNS resolution.

**Comment by jwbonner** [comment]
My best guess is that Windows isn't handling the domain name as well as Linux and macOS (I'm not an expert on this, but domain name resolution on Windows tends to be a little odd). I'll do some more testing on Windows devices and get back to you.
